8Knot/pages/index/search_utils.py

```diff
@@ -0,0 +1,248 @@
+"""
+Search utilities for the 8Knot application.
+Provides improved search algorithms for the searchbar.
+"""
+from typing import List, Dict, Any
+import re
+from difflib import SequenceMatcher
+
+def levenshtein_distance(s1: str, s2: str) -> int:
```

Comment:
Is there a reason you are implementing this vs using a package? https://pypi.org/project/python-Levenshtein/